internal/injector/parse.go

```diff
@@ -0,0 +1,137 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package injector
+
+import (
+	"bytes"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io"
+	"os"
+)
+
+// parseFiles parses all provided files, after having applied any leading
+// "//line" directives if present.
+func (i *Injector) parseFiles(fset *token.FileSet, files []string) ([]*ast.File, error) {
+	result := make([]*ast.File, len(files))
+	for idx, file := range files {
+		var err error
+		result[idx], err = i.parseFile(fset, file)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return result, nil
+}
```

Comment:
```suggestion
func parseFiles(fset *token.FileSet, files []string) ([]*ast.File, error) {
	result := make([]*ast.File, len(files))
	for idx, file := range files {
		var err error
		result[idx], err = parseFile(fset, file)
		if err != nil {
			return nil, err
		}
	}
	return result, nil
}
```
`parseFiles` and `parseFile` do not use the injector so there is no need for them to be methods